### PR TITLE
[GLUTEN-6590][CH] Support compact mergetree file on s3

### PR DESCRIPTION
**Pull request by lwz9103**
## What changes were proposed in this pull request?

(Fixes: \#6590)

## How was this patch tested?

unit tests

### CODE REVIEW

File: backends-clickhouse/src/test/scala/org/apache/gluten/execution/GlutenClickHouseMergeTreeWriteOnS3Suite.scala [hunk 1]
```diff
@@ -635,6 +671,7 @@ class GlutenClickHouseMergeTreeWriteOnS3Suite
 
     withSQLConf(
       "spark.databricks.delta.optimize.minFileSize" -> "200000000",
+      "spark.gluten.sql.columnar.backend.ch.runtime_settings.mergetree.insert_without_local_storage" -> "true",
```
Comment (liuneng1994) [inline backends-clickhouse/src/test/scala/org/apache/gluten/execution/GlutenClickHouseMergeTreeWriteOnS3Suite.scala:674]: why add this config

File: cpp-ch/local-engine/Common/CHUtil.cpp [hunk 2]
```diff
@@ -556,7 +556,7 @@ std::vector<String> BackendInitializerUtil::wrapDiskPathConfig(
     if (path_prefix.empty() && path_suffix.empty())
         return changed_paths;
     Poco::Util::AbstractConfiguration::Keys disks;
-    std::unordered_set<String> disk_types = {"s3", "hdfs_gluten", "cache"};
+    std::unordered_set<String> disk_types = {"s3_gluten", "hdfs_gluten", "cache"};
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/CHUtil.cpp:559]: GlutenClickHouseWholeStageTransformerSuite configuration of disk type for UT should be changed

**Review by liuneng1994 — CHANGES_REQUESTED** [review]
Please check unit tests can test current codes

File: backends-clickhouse/src/test/scala/org/apache/gluten/execution/GlutenClickHouseMergeTreeWriteOnS3Suite.scala (hunk 1 above)
Comment (lwz9103) [reply backends-clickhouse/src/test/scala/org/apache/gluten/execution/GlutenClickHouseMergeTreeWriteOnS3Suite.scala:674]: Test directly merge parts on s3. See this code branch
image

File: cpp-ch/local-engine/Common/CHUtil.cpp (hunk 2 above)
Comment (lwz9103) [reply cpp-ch/local-engine/Common/CHUtil.cpp:559]: Disk type is already set to s3_gluten.
image

**Comment by lwz9103** [comment]
> Please check unit tests can test current codes

Please see this code to test compact files on s3
image

**Review by liuneng1994 — APPROVED** [review]

**Comment by lwz9103** [comment]
### Test Result
Compact files on s3
image
Local UT test
image

**Pull request merged**